x-pack/plugins/security/server/saved_objects/audit_events.ts

```diff
@@ -0,0 +1,149 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import { AuditEventDecorator } from '../../../../../src/core/server';
+
+export interface SavedObjectEventArgs {
+  action: string;
+  objects: ReadonlyArray<{
+    type: string;
+    id?: string;
+    namespaces?: string[];
+  }>;
+  error?: Error;
+}
+
+export const savedObjectCreateEvent: AuditEventDecorator<SavedObjectEventArgs> = (
+  event,
+  { action, objects, error }
+) => {
+  const doc =
+    objects.length === 1
+      ? `saved object '${objects[0].id}' of type '${objects[0].type}'`
+      : `${objects.length} saved objects`;
+
+  return {
+    ...event,
+    message: error
+      ? `Failed attempt to create ${doc} by user '${event.user.name}'`
+      : `User '${event.user.name}' created ${doc}`,
+    event: {
+      action,
+      category: 'database',
+      type: 'creation',
+      outcome: error ? 'failure' : 'success',
+    },
+    error: error
+      ? {
+          code: error.name,
+          message: error.message,
+        }
+      : undefined,
+    kibana: {
+      namespace: event.kibana.namespace,
```

Comment:
Yeah. Maybe a simple spread would be more future-proof here?

```suggestion
      ...event.kibana,
```

Reply:
This was just to avoid yet another spread operator (since namespace is the only property provided so going to be faster to assign rather than spread) but happy to change this.